### PR TITLE
Fix hack for inverting dFdy

### PR DESCRIPTION
**Pull request by bghgary**
Use syntax tree from glslang instead of a string replace to invert dFdy operand.

### CODE REVIEW

File: Library/Source/ShaderCompiler.cpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+                        unary->setOperand(intermediate->addUnaryNode(glslang::EOpNegative, unary->getOperand(), {}));
+                        return false;
+                    }
+                }
+
+                return true;
+            }
+
+        private:
+            glslang::TIntermediate* intermediate;
```
Comment (syntheticmagus) [inline Library/Source/ShaderCompiler.cpp]: nit: m_intermediate?

File: Library/Source/ShaderCompiler.cpp [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+#include <glslang/MachineIndependent/localintermediate.h>
+
+namespace babylon
+{
+    namespace
+    {
+        class InvertYDerivativeOperandsTraverser : public glslang::TIntermTraverser
+        {
+        public:
+            InvertYDerivativeOperandsTraverser(glslang::TIntermediate* intermediate) : intermediate(intermediate)
```
Comment (bghgary) [inline Library/Source/ShaderCompiler.cpp]:
```suggestion
            InvertYDerivativeOperandsTraverser(glslang::TIntermediate* intermediate) : m_intermediate(intermediate)
```

File: Library/Source/ShaderCompiler.cpp (hunk 1 above)
Comment (bghgary) [reply Library/Source/ShaderCompiler.cpp]:
```suggestion
            glslang::TIntermediate* m_intermediate;
```

File: Library/Source/ShaderCompiler.cpp [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            }
+
+            virtual bool visitUnary(glslang::TVisit visit, glslang::TIntermUnary* unary) override
+            {
+                if (visit == glslang::EvPreVisit)
+                {
+                    auto op = unary->getOp();
+                    if (op == glslang::EOpDPdy || op == glslang::EOpDPdyFine || op == glslang::EOpDPdyCoarse)
+                    {
+                        unary->setOperand(intermediate->addUnaryNode(glslang::EOpNegative, unary->getOperand(), {}));
```
Comment (bghgary) [inline Library/Source/ShaderCompiler.cpp]:
```suggestion
                        unary->setOperand(m_intermediate->addUnaryNode(glslang::EOpNegative, unary->getOperand(), {}));
```

**Pull request merged**